x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Page.java

```diff
@@ -222,6 +221,43 @@ public void writeTo(StreamOutput out) throws IOException {
      */
     public void releaseBlocks() {
         blocksReleased = true;
-        Releasables.closeExpectNoException(blocks);
+        // blocks can be used as multiple columns
+        var map = new IdentityHashMap<Block, Object>(mapSize(blocks.length));
+        var DUMMY = new Object();
+        for (Block b : blocks) {
+            if (map.putIfAbsent(b, DUMMY) == null) {
+                Releasables.closeExpectNoException(b);
+            }
+        }
```

Comment:
I think this improves the issue, but there still can be problems because non-identical blocks can be backed by the same vector or array.

Additionally, we create and populate a new hash map on each page release; that's probably quite expensive.

We can merge this to fix the problem right now, but IMO should mark this with a todo comment to remove this logic once possible. We want ref counting, this will fix the problem more idiomatically.